projects/plugins/jetpack/modules/widget-visibility/widget-conditions.php

```diff
@@ -1148,8 +1148,16 @@ public static function migrate_post_type_rules() {
 					continue;
 				}
 
-				$opts      = $wp_registered_widgets[ $widget ];
-				$instances = get_option( $opts['callback'][0]->option_name );
+				$opts = $wp_registered_widgets[ $widget ];
+
+				$id_base       = wp_parse_widget_id( $widget )['id_base'];
+				$widget_object = $wp_widget_factory->get_widget_object( $id_base );
+
+				if ( ! $widget_object ) {
+					continue;
+				}
+
+				$instances = get_option( $widget_object->option_name );
```

Comment:
We missed one, there's an `update_option` call below that needs changing from `$opts['callback'][0]` to `$widget_object` too https://github.com/Automattic/jetpack/blob/83d3a300ec48ac22c432d51fe11e3d226f4ebf27/projects/plugins/jetpack/modules/widget-visibility/widget-conditions.php#L1203

Reply:
Ouch, that would have been painful to leave in, thanks!